pymongo/database.py

```diff
@@ -620,7 +620,7 @@ def command(
         session: Optional["ClientSession"] = None,
         comment: Optional[Any] = None,
         **kwargs: Any,
-    ) -> Dict[str, Any]:
+    ) -> _DocumentOut:
```

Comment:
Is there a simple way to make this method generic such that it returns the type of codec_options.document_class?

Reply:
We had to pick one source of truth, which was the document type of the client given to the constructor.  Technically the `codec_options` given to the constructor any of the methods can override it, but it was too difficult to express as a type.  We had agreed that the most common case is to specify it once in the client, and the user can cast if they are using explicit `codec_options` in other places.

Reply:
db.command() doesn't use the client's or the database's document_class. It only uses the document_class from the codec_options argument.

Reply:
Understood now, thanks

Reply:
Drat, this is a problem: https://bugs.python.org/issue43923.  We can't have a generic namedtuple.

Reply:
We could explicitly add the properties or add the appropriate `__getattr__`.  The downside would be that `isinstance` would fail if anyone were relying on that.

Reply:
In that case I'm okay with having db.command (somewhat incorrectly) return _DocumentOut. We can open a new issue to support generics and have it depend on that cpython bug.

Reply:
Done and done: https://jira.mongodb.org/browse/PYTHON-3143